### ISSUE TITLE
2.0.4 - WYSIWYG Editor strips empty HTML codes out 

### ISSUE THREAD

**Issue by grasdaggel** [magento/magento2#4324]
We are having an issue with the WYSIWYG editor where we need to add something like 
`<em class="vendor-icon vendor-cart"> </em>`
Which would result in an icon

The editor strips out empty html tags and therefore we can't add Icons to our page content.

This can't be really intended in views of technologies like Fontello, Font Awesome and stuff like that.

Whats a good workaround for this? We can't really deactivate the editor, though (customer needs it)

**Comment by BenGlassAtlas** [comment]
I think this is down to Tiny MCE editor and not so much Magento 2, I'm not sure how helpful it will be but this is the workaround for Magento 1 - http://alanstorm.com/magento_html5_tinymce 

A quick workaround worth trying would be to add a span with content that has display: none set. Something like:

``` html
<em class="vendor-icon vendor-cart"><span style"display: none;">1</span></em>
```

**Comment by shiftedreality** [comment]
Hi @kingalca 

Thank you for reporting this issue. 
I created internal ticket MAGETWO-52378 to investigate and fix.

**Comment by piotrekkaminski** [comment]
cc @elenleonova

**Comment by LiliSpace48** [comment]
@shiftedreality @elenleonova H,  is there any update about this issue please? thanks

**Comment by LiliSpace48** [comment]
@vkorotun @elenleonova  Hello, could you let me know if there is any update on your internal issue MAGETWO-52378  please.
Thanks

**Comment by hbhardik** [comment]
Hello, is there any update for the internal issue ?

**Comment by olysenko** [comment]
MAGETWO-52378 is delivered into develop branch

**Comment by Ctucker9233** [comment]
@magento-engcom-team This fix is not included in 2.1.10. What release might it be in?

**Comment by hostep** [comment]
@Ctucker9233: looks like the fixes are only included in 2.2.0 and up
Relevant commits, which seem to have something to do with this: https://github.com/magento/magento2/commit/f1ebd5c2bdf07c114f24126be8c4e97c0149997c & https://github.com/magento/magento2/commit/9a998de390ee5bf0608577c666a12bf292493242

Feel free to backport these to 2.1.x if you have some spare time for this.

**Comment by Ctucker9233** [comment]
@hostep Alright, I'll look at it and give it a go.
